internal/test/framework/windowsvm.go

```diff
@@ -51,57 +52,117 @@ type TestWindowsVM interface {
+// waitForMachines() waits until all the machines required are in Provisioned state. It returns an array of all
+// the machines created. All the machines are created concurrently.
+func (f *TestFramework) waitForMachines(vmCount int) ([]mapi.Machine, error) {
+	log.Print("Waiting for Machine Sets ")
+	windowsOSLabel := "machine.openshift.io/os-id"
+	var provisionedMachines []mapi.Machine
+	// it takes approximately 12 minutes in the CI for all the machines to appear.
+	timeOut := 12 * time.Minute
+	startTime := time.Now()
+	for i := 0; time.Since(startTime) <= timeOut; i++ {
+		allMachines := &mapi.MachineList{}
+
+		err := f.client.List(context.TODO(), allMachines, client.InNamespace("openshift-machine-api"), client.HasLabels{windowsOSLabel})
 		if err != nil {
-			return nil, fmt.Errorf("error creating Windows VM: %v", err)
-		}
-		// TypeAssert to the WindowsVM struct we want
-		winVM, ok := windowsVM.(*types.Windows)
-		if !ok {
-			return nil, fmt.Errorf("error asserting Windows VM: %v", err)
+			return nil, fmt.Errorf("failed to list machines: %v", err)
 		}
-		w.Windows = winVM
-	} else {
-		//TODO: Add username as well, as it will change depending on cloud provider
-		if credentials.GetIPAddress() == "" || credentials.GetPassword() == "" {
-			return nil, fmt.Errorf("password or IP address not specified in credentials")
+		provisionedMachines = []mapi.Machine{}
+
+		phaseProvisioned := "Provisioned"
+
+		for _, machine := range allMachines.Items {
+			instanceStatus := machine.Status
+			if instanceStatus.Phase == nil {
+				continue
+			}
+			instancePhase := *instanceStatus.Phase
+			if instancePhase == phaseProvisioned {
+				provisionedMachines = append(provisionedMachines, machine)
+			}
 		}
-		w.Windows = &types.Windows{}
-		w.Credentials = credentials
+		time.Sleep(5 * time.Second)
 	}
-
-	if err := w.SetupWinRMClient(); err != nil {
-		return w, fmt.Errorf("failed to setup winRM client for the Windows VM: %v", err)
+	if vmCount == len(provisionedMachines) {
+		return provisionedMachines, nil
 	}
-	// Wait for some time before starting configuring of ssh server. This is to let sshd service be available
-	// in the list of services
-	// TODO: Parse the output of the `Get-Service sshd, ssh-agent` on the Windows node to check if the windows nodes
-	// has those services present
-	if !skipSetup {
-		time.Sleep(time.Minute)
-		if err := w.ConfigureOpenSSHServer(); err != nil {
-			return w, fmt.Errorf("failed to configure OpenSSHServer on the Windows VM: %v", err)
-		}
+	return nil, fmt.Errorf("expected event count %d but got %d", vmCount, len(provisionedMachines))
+}
+
+// newWindowsVM creates and sets up a Windows VM in the cloud and returns the WindowsVM interface that can be used to
+// interact with the VM. If no error is returned then it is guaranteed that the VM was
+// created and can be interacted with.
+func (f *TestFramework) newWindowsVM(vmCount int) ([]TestWindowsVM, error) {
```

Comment:
`newWindowsVM` implies one VM while you are creating `vmCount` number of them. This should rather be `newWindowsMachineSet`.